api/incubator/src/main/java/io/opentelemetry/api/incubator/config/InstrumentationConfigUtil.java

```diff
@@ -0,0 +1,143 @@
+          String service = entry.getString("service");
+          if (peer != null && service != null) {
+            serviceMapping.put(peer, service);
+          }
+        });
+    return serviceMapping.isEmpty() ? null : serviceMapping;
+  }
+
+  /**
+   * Return {@code .instrumentation.general.http.client.request_captured_headers}, or null if none
+   * is configured.
+   *
+   * @throws DeclarativeConfigException if an unexpected type is encountered accessing the property
+   */
+  @Nullable
+  public static List<String> httpClientRequestCapturedHeaders(ConfigProvider configProvider) {
+    return getOrNull(
+        configProvider,
+        config -> config.getScalarList("request_captured_headers", String.class),
+        "general",
+        "http",
+        "client");
+  }
+
+  /**
+   * Return {@code .instrumentation.general.http.client.response_captured_headers}, or null if none
+   * is configured.
+   *
+   * @throws DeclarativeConfigException if an unexpected type is encountered accessing the property
+   */
+  @Nullable
+  public static List<String> httpClientResponseCapturedHeaders(ConfigProvider configProvider) {
+    return getOrNull(
+        configProvider,
+        config -> config.getScalarList("response_captured_headers", String.class),
+        "general",
+        "http",
+        "client");
+  }
+
+  /**
+   * Return {@code .instrumentation.general.http.server.request_captured_headers}, or null if none
+   * is configured.
+   *
+   * @throws DeclarativeConfigException if an unexpected type is encountered accessing the property
+   */
+  @Nullable
+  public static List<String> httpServerRequestCapturedHeaders(ConfigProvider configProvider) {
+    return getOrNull(
+        configProvider,
+        config -> config.getScalarList("request_captured_headers", String.class),
+        "general",
+        "http",
+        "server");
+  }
+
+  /**
+   * Return {@code .instrumentation.general.http.server.response_captured_headers}, or null if none
+   * is configured.
+   *
+   * @throws DeclarativeConfigException if an unexpected type is encountered accessing the property
+   */
+  @Nullable
+  public static List<String> httpSeverResponseCapturedHeaders(ConfigProvider configProvider) {
+    return getOrNull(
+        configProvider,
+        config -> config.getScalarList("response_captured_headers", String.class),
+        "general",
+        "http",
+        "server");
+  }
+
+  /**
+   * Return {@code .instrumentation.java.<instrumentationName>}, or null if none is configured.
+   *
+   * @throws DeclarativeConfigException if an unexpected type is encountered accessing the property
+   */
+  @Nullable
+  public static DeclarativeConfigProperties javaInstrumentationConfig(
+      ConfigProvider configProvider, String instrumentationName) {
+    return getOrNull(configProvider, config -> config.getStructured(instrumentationName), "java");
+  }
+
+  @Nullable
+  private static <T> T getOrNull(
```

Comment:
this seems to be an important method that should be added to the api itself

the segments vararg could be the last parameter of each method in the api

Reply:
I could be convinced to add promote it to ConfigProvider as syntactic sugar. For now, will make it public so callers can at least access it.